packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/SliceCard.tsx

```diff
@@ -1,56 +1,70 @@
+import { Button } from "@prismicio/editor-ui";
+
 import { Card, CardFooter, CardMedia } from "@/components/Card";
 
 interface SliceCardProps {
   slice: Slice;
+  onRetry?: () => void;
 }
 
 export function SliceCard(props: SliceCardProps) {
-  const { slice } = props;
+  const { slice, onRetry } = props;
+
+  const loading = slice.status === "uploading";
+  const error = slice.status === "uploadError";
 
   return (
-    <Card disabled={slice.status === "loading"} style={{ width: 394 }}>
+    <Card disabled={loading} style={{ width: 394 }}>
       {slice.status === "success" ? (
         <CardMedia src={slice.thumbnailUrl} />
       ) : (
         <CardMedia component="div" />
       )}
       <CardFooter
-        loading={slice.status === "loading"}
+        loading={loading}
         startIcon={getStartIcon(slice.status)}
         title={slice.image.name}
         subtitle={getSubtitle(slice.status)}
+        error={error}
+        action={
+          error ? (
+            <Button startIcon="refresh" color="grey" onClick={onRetry}>
+              Retry
+            </Button>
+          ) : undefined
+        }
       />
     </Card>
   );
 }
 
-export type Slice =
-  | { status: "loading"; image: File }
+export type Slice = { image: File } & (
+  | { status: "uploading" }
+  | { status: "uploadError" }
```

Comment:
```suggestion
  | { status: "uploadError", onRetry: ... }
```

Reply:
Ah okay, I see now. This is better than what I was trying to do. Thanks, it's done ☑️